CSI Snapshot tests are failing in master

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The snapshot tests are failing in master in the helm suite for the past couple days (since build 2439). Note that the helm suite runs on Ceph nautilus 14.2.12. I might have suspected it was related to CSI 3.2.0, but the tests are not failing in the release branch.

From the [build log](https://jenkins.rook.io/blue/rest/organizations/jenkins/pipelines/rook/pipelines/rook/branches/master/runs/2439/nodes/63/steps/132/log/?start=0):
```
[2020-12-14T21:30:39.516Z] --- FAIL: TestCephHelmSuite (738.05s)
[2020-12-14T21:30:39.516Z]     --- PASS: TestCephHelmSuite/TestARookInstallViaHelm (0.04s)
[2020-12-14T21:30:39.516Z]     --- FAIL: TestCephHelmSuite/TestBlockStoreOnRookInstalledViaHelm (265.13s)
[2020-12-14T21:30:39.516Z]         ceph_base_block_test.go:194: 
[2020-12-14T21:30:39.516Z]             	Error Trace:	ceph_base_block_test.go:194
[2020-12-14T21:30:39.516Z]             	            				ceph_base_block_test.go:433
[2020-12-14T21:30:39.516Z]             	            				ceph_helm_test.go:99
[2020-12-14T21:30:39.516Z]             	Error:      	Received unexpected error:
[2020-12-14T21:30:39.516Z]             	            	giving up waiting for "rbd-pvc-snapshot" snapshot in namespace "default"
[2020-12-14T21:30:39.516Z]             	Test:       	TestCephHelmSuite/TestBlockStoreOnRookInstalledViaHelm
```

```
[2020-12-14T21:30:39.516Z]     --- FAIL: TestCephHelmSuite/TestFileStoreOnRookInstalledViaHelm (31.74s)
[2020-12-14T21:30:39.516Z]         ceph_base_file_test.go:122: 
[2020-12-14T21:30:39.516Z]             	Error Trace:	ceph_base_file_test.go:122
[2020-12-14T21:30:39.516Z]             	            				ceph_base_file_test.go:373
[2020-12-14T21:30:39.516Z]             	            				ceph_helm_test.go:104
[2020-12-14T21:30:39.516Z]             	Error:      	Received unexpected error:
[2020-12-14T21:30:39.516Z]             	            	Failed to run: kubectl [create -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/v3.0.0/client/config/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml] : exit status 1
[2020-12-14T21:30:39.516Z]             	Test:       	TestCephHelmSuite/TestFileStoreOnRookInstalledViaHelm
```

**Expected behavior:**
Tests should pass consistently.

**How to reproduce it (minimal and precise):**
<!-- Please let us know any circumstances for reproduction of your bug. -->

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

--- a/tests/framework/utils/k8s_helper.go
+++ b/tests/framework/utils/k8s_helper.go
@@ -22,6 +22,8 @@ import (
 	"encoding/json"
 	"fmt"
 	"html/template"
+	"io/ioutil"
+	"net/http"
 	"os"
 	"path"
 	"strconv"
@@ -30,13 +32,13 @@ import (
 	"time"
 
 	"github.com/coreos/pkg/capnslog"
+	"github.com/pkg/errors"
 	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
 	"github.com/rook/rook/pkg/clusterd"
 	"github.com/rook/rook/pkg/util/exec"
 	"github.com/stretchr/testify/require"
 	apps "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
@@ -164,7 +166,7 @@ func (k8sh *K8sHelper) KubectlWithStdin(stdin string, args ...string) (string, e
 	if cmdOut.ExitCode != 0 {
 		k8slogger.Errorf("Failed to execute stdin: %s %v : %v", cmd, args, cmdOut.Err.Error())
 		if strings.Contains(cmdOut.Err.Error(), "(NotFound)") || strings.Contains(cmdOut.StdErr, "(NotFound)") {
-			return cmdOut.StdErr, errors.NewNotFound(schema.GroupResource{}, "")
+			return cmdOut.StdErr, kerrors.NewNotFound(schema.GroupResource{}, "")
 		}
 		return cmdOut.StdErr, fmt.Errorf("Failed to run stdin: %s %v : %v", cmd, args, cmdOut.StdErr)
 	}
@@ -174,6 +176,23 @@ func (k8sh *K8sHelper) KubectlWithStdin(stdin string, args ...string) (string, e
 
 	return cmdOut.StdOut, nil
 
+}
+
+func getManifestFromURL(url string) (string, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to get manifest from url %s", url)
+	}
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to read manifest from url %s", url)
+	}
+	return string(body), nil
 }
 
 func getKubeConfig(executor exec.Executor) (*rest.Config, error) {
@@ -372,7 +391,7 @@ func (k8sh *K8sHelper) WaitForCustomResourceDeletion(namespace string, checkerFu
 			time.Sleep(2 * time.Second)
 			continue
 		}
-		if errors.IsNotFound(err) {
+		if kerrors.IsNotFound(err) {
 			logger.Infof("custom resource %s deleted", namespace)
 			return nil
 		}
@@ -411,7 +430,7 @@ func (k8sh *K8sHelper) CreateNamespace(namespace string) error {
 	ctx := context.TODO()
 	ns := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
 	_, err := k8sh.Clientset.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
-	if err != nil && !errors.IsAlreadyExists(err) {
+	if err != nil && !kerrors.IsAlreadyExists(err) {
 		return fmt.Errorf("failed to create namespace %s. %+v", namespace, err)
 	}
 
@@ -423,7 +442,7 @@ func (k8sh *K8sHelper) CountPodsWithLabel(label string, namespace string) (int, 
 	options := metav1.ListOptions{LabelSelector: label}
 	pods, err := k8sh.Clientset.CoreV1().Pods(namespace).List(ctx, options)
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if kerrors.IsNotFound(err) {
 			return 0, nil
 		}
 		return 0, err
@@ -507,7 +526,7 @@ func (k8sh *K8sHelper) WaitUntilPodWithLabelDeleted(label string, namespace stri
 	ctx := context.TODO()
 	for i := 0; i < RetryLoop; i++ {
 		pods, err := k8sh.Clientset.CoreV1().Pods(namespace).List(ctx, options)
-		if errors.IsNotFound(err) {
+		if kerrors.IsNotFound(err) {
 			logger.Infof("error Found err %v", err)
 			return true
 		}
@@ -953,7 +972,7 @@ func (k8sh *K8sHelper) isVolumeExist(namespace, name string) (bool, error) {
 	ctx := context.TODO()
 	_, err := k8sh.RookClientset.RookV1alpha2().Volumes(namespace).Get(ctx, name, metav1.GetOptions{})
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if kerrors.IsNotFound(err) {
 			return false, nil
 		}
 		return false, err
@@ -1276,7 +1295,7 @@ func (k8sh *K8sHelper) WaitUntilPodIsDeleted(name, namespace string) bool {
 	ctx := context.TODO()
 	for i := 0; i < RetryLoop; i++ {
 		_, err := k8sh.Clientset.CoreV1().Pods(namespace).Get(ctx, name, metav1.GetOptions{})
-		if err != nil && errors.IsNotFound(err) {
+		if err != nil && kerrors.IsNotFound(err) {
 			return true
 		}
 
@@ -1408,7 +1427,7 @@ spec:
   type: NodePort
 `
 	_, err := k8sh.KubectlWithStdin(externalSvc, []string{"apply", "-f", "-"}...)
-	if err != nil && !errors.IsAlreadyExists(err) {
+	if err != nil && !kerrors.IsAlreadyExists(err) {
 		return fmt.Errorf("failed to create external service. %+v", err)
 	}
 

--- a/tests/framework/utils/snapshot.go
+++ b/tests/framework/utils/snapshot.go
@@ -17,6 +17,7 @@ import (
 	"context"
 	"fmt"
 	"strconv"
+	"strings"
 	"time"
 
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
@@ -58,8 +59,15 @@ func (k8sh *K8sHelper) CheckSnapshotISReadyToUse(name, namespace string, retries
 
 // snapshotController creates or deletes the snapshotcontroller and required RBAC
 func (k8sh *K8sHelper) snapshotController(action string) error {
-	controller := fmt.Sprintf("%s/%s/%s", repoURL, snapshotterVersion, controllerPath)
-	_, err := k8sh.Kubectl(action, "-f", controller)
+	controllerURL := fmt.Sprintf("%s/%s/%s", repoURL, snapshotterVersion, controllerPath)
+	controllerManifest, err := getManifestFromURL(controllerURL)
+	if err != nil {
+		return err
+	}
+	controllerManifest = strings.Replace(controllerManifest, "canary", snapshotterVersion, -1)
+	logger.Infof("snapshot controller: %s", controllerManifest)
+
+	_, err = k8sh.KubectlWithStdin(controllerManifest, action, "-f", "-")
 	if err != nil {
 		return err
 	}

